Makefile

```diff
@@ -15,7 +15,7 @@ endif
 # Override this with any scopes for testing/benching.
 export SCOPE ?= ""
 # Override this with any extra flags for cargo bench
-export CARGO_BENCH_FLAGS ?= ""
```

Comment:
Would suggest either leaving the `""` here or also removing it from the other `?= ""` expressions.

(Personally, prefer the `""` to indicate that it accepts a string.)

Reply:
The problem with this is that when the value is absent the literal `""` is passed.

Reply:
Does that lead to unintended behavior in our case? Is it because `argc` is increased but only an empty string being passed?

Reply:
Yes, it breaks things. For instance when the `SCOPE` has the value `""` we need some special handling if we want to omit it. I suspect people didn't notice this in other places because they don't properly escape things in bash (which we definitely should do).

Here's an example of a workaround for this problem I employed some time ago: https://github.com/timberio/vector/blob/594a6c43e4e0f2d4e01136618f85a3a782eeec14/scripts/test-e2e-kubernetes.sh#L135

Reply:
I see, thanks! Would you mind updating the other `?= ""` statements in this PR or a follow-up?

Reply:
I think it might unwind into fixing a bunch of scripts that somehow oddly rely on this, so it can result in a non-trivial amount of work.
Also, I presume the only way to really address this issue is to educate people on proper `bash` usage. I've tried to introduce `shellcheck` already, and it somewhat helps. My concern is, while there is a lot of material on `bash` out there, people just don't care, as they see `bash` as a way to hack things quickly, and don't really want to learn it beyond the basics. Same applies to `make` and `Makefile` syntax.

The approach of leading the motion by example also didn't give good results (see my attempt at #2482).
In short, I'm thinking about new approach to step up the quality of our scripting setup, but didn't come us with a solution yet.

So I tend to keep myself from working on such things - where the amount of work seems non-trivial - without a proper issue/scheduling. Creating an issue to eventually address this is a good idea though!

Reply:
I can certainly understand (and share) the pain of trying to improve on bash, with the incredibly many pitfalls. It feels like a book of trivia to memorize, with little chance of transfering that knowledge. I wouldn't put too much blame on the people, but you're right that it's an essential tool to know because of its prevalence.

`shellcheck` looks like a good measure.

Opened an issue for this one: https://github.com/timberio/vector/issues/6336.

Reply:
Thanks!